src/core/cgroup.c

```diff
@@ -4858,12 +4863,14 @@ static int unit_get_io_accounting_raw(
 int unit_get_io_accounting(
                 Unit *u,
                 CGroupIOAccountingMetric metric,
-                bool allow_cache,
                 uint64_t *ret) {
 
         uint64_t raw[_CGROUP_IO_ACCOUNTING_METRIC_MAX];
         int r;
 
+        assert(metric < _CGROUP_IO_ACCOUNTING_METRIC_MAX
+               || (ret == NULL && metric == _CGROUP_IO_ACCOUNTING_METRIC_INVALID));
```

Comment:
Hmm, the `< _CGROUP_IO_...MAX` part would always be true, no? Maybe:

```c
/* We specialize the case where _..._INVALID is specified as metric, and no retval is expected, which means the caller wants to refresh the cached values */
assert(metric >= 0 || (metric == _..._INVALID && !ret));
assert(metric < _..._MAX);
```

Reply:
Oops, that's a thinko — I somehow ended up convincing myself that _INVALID is > _MAX rather than being -EINVAL. Fixed.

Reply:
Please include a comment like this. When doing null ptr checks we rely on them degrading to bool BTW, no need for == NULL

Reply:
OK. The coding style said that I "may" use downgrade-to-bool for NULL checks, so I went with my own preference, but if it's actually a requirement, fine.